### ISSUE TITLE
Converted PyTorch to Keras without error, but inference results are different

### ISSUE THREAD

**Issue by netw0rkf10w** [microsoft/MMdnn#318]
Hello,

I have converted a ResNet-152 model that I trained on a custom dataset from PyTorch to Keras, but it seems the conversion was not successful because when testing on an image, the output of the two networks are different.

Here's how I performed the conversion:

1. Convert PyTorch to IR:
`mmtoir -f pytorch -d out --inputShape 3 224 224 -n resnet152_pytorch.pth`
Three files are produced: `out.json`, `out.pb` and `out.npy`.

2. Convert IR to Keras code snippet:
`mmtocode -f keras --IRModelPath out.pb --dstModelPath resnet152_keras.py`

3. Save the weights to Keras .h5:
`python -m mmdnn.conversion.examples.keras.imagenet_test -n resnet152_keras.py -w out.npy --dump resnet152_keras.h5`

And here's how I tested them:

1. First read an image, resize it and convert to float format (it's just for testing so we don't need proper pre-processing):
    
   ```
    image = cv2.imread(image_path, 1)
    image = cv2.resize(image, (224, 224), interpolation = cv2.INTER_CUBIC)
    image = image.astype(float)
    ```
   
2. Load the model and do inference for PyTorch:

        device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
        model = torch.load('resnet152_pytorch.pth')
        image_cwh = np.rollaxis(image, 2, 0) # convert the image to CWH format
        im_tensor = torch.from_numpy(image_cwh)
        im_tensor.unsqueeze_(0) # add batch dimension

        # send the input the and the model to GPU device
        im_tensor = im_tensor.to(device)
        model = model.to(device)

        # prediction
        output = model(im_tensor)
        print(output)

3. Load the model and do inference for Keras:
      
        model = keras.models.load_model('resnet152_keras.h5')
        output = model.predict_on_batch(np.expand_dims(image, axis=0))
        print(output)

And this is what I obtained:    

> Pytorch: tensor([[-0.7422, -0.9134, -0.5829,  ..., -0.0954, -1.2378,  0.0789]], device='cuda:0')
> Keras: [[ -52.964676 -101.15237   -71.54291  ...   20.734167  -69.53998  -58.768787]]

Thank you very much in advance for your help!

**Comment by kitstar** [comment]
Hi @netw0rkf10w , please apply the pytorch preprocess method to the keras model and try if it works.

**Comment by netw0rkf10w** [comment]
@kitstar As you can see from the above code, there's only a difference in the pre-processing between PyTorch and Keras, that is the channel order of the image. PyTorch requires CWH while Keras requires HWC. This is supposed to be taken into account by mmdnn already, right?

**Comment by kitstar** [comment]
@netw0rkf10w . The channel problem is taken in MMdnn, and offical pytorch resnet 152 -> keras is tested. Not quite sure about the problem. Could you share the model for testing? Thanks.

**Comment by netw0rkf10w** [comment]
@kitstar Sure. Here it is: https://drive.google.com/file/d/1Z393bCEKN1kMKI_x_nuccurIWUp2_9rg/view?usp=sharing

> offical pytorch resnet 152 -> keras is tested

You mean we can convert PyTorch ResNet152 to Keras using a one-line command? Like this: `mmconvert -sf pytorch -in resnet152_pytorch.pth -df keras -om resnet152_keras.h5`?

**Comment by kitstar** [comment]
@netw0rkf10w yes it is, with mmdownload to download the pytorch resnet152 models.

**Comment by CyanideBoy** [comment]
@netw0rkf10w Did you find out what's wrong?

**Comment by netw0rkf10w** [comment]
@CyanideBoy Sorry it was long time ago and I have no idea what I did.
